### DISCUSSION TITLE
bypass (row level security) RLS in a postgres function

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3563] [Questions] [answered]
Hi!
I have a `users` table, with some RLS with `auth.uid.()` table!
I have a function that check if the email already exist or not - in my table 'users'.
So when i run it with `rpc` it's gives me always the same result `return false`, So how i can ran this `rpc` function and bypass the RLS in the client?

> i know i can use `service_role` project API key with the server side - but my web app currently works in the client side only.

Thanks!

**Comment by soedirgo** [comment, marked as answer]
Hi Zakaria, you can use `SECURITY DEFINER` functions like [here](https://supabase.io/docs/guides/auth/row-level-security#policies-with-security-definer-functions). If you are using the dashboard to create the function, you can enable the option from advanced settings.

**Reply by zakaria-chahboun** [reply]
@soedirgo Thanks!

**Reply by al-shseo** [reply]
I spent a lot of time on this
I didn't even know there was an option like that.

Thanks for solving it! thank you! 😊
